packages/shared/src/components/buttons/ButtonV2.spec.tsx

```diff
@@ -0,0 +1,120 @@
+import { render, RenderResult, screen } from '@testing-library/react';
+import React, { HTMLAttributes } from 'react';
+import {
+  Button,
+  ButtonProps,
+  ButtonColor,
+  ButtonKind,
+  ButtonVariant,
+  ButtonIconPosition,
+  AllowedTags,
+} from './ButtonV2';
+import UpvoteIcon from '../icons/Upvote';
+
+const renderComponent = <Tag extends AllowedTags>(
+  props: Partial<ButtonProps<Tag>> = {},
+): RenderResult => {
+  return render(<Button {...props} />);
+};
+
+it('should render children', async () => {
```

Comment:
Should we maybe wrap these sections in a `describe`? Just thinking of making it easier to isolate related tests. Not required just a thought.

For example we will have a `describe('button variants', () => { // tests go here })`

Reply:
I tried, please take a look again? :)